test/test_profiler.py

```diff
@@ -163,6 +165,129 @@ def test_kineto_multigpu(self):
         self.assertTrue(found_gemm_1)
         self.assertTrue(found_cuda)
 
+    def test_memory_profiler(self):
+        def run_profiler(tensor_creation_fn, metric):
+            # collecting allocs / deallocs
+            with _profile(profile_memory=True, record_shapes=True, use_kineto=kineto_available()) as prof:
```

Comment:
we should make a follow-up item to either migrate or duplicate the tests with `profiler` instead of `autograd.profiler`.

Reply:
all our autograd.profiler tests should use `use_kineto=kineto_available()` now, and we have both kineto-enabled and not enabled CI builds; eventually when autograd.profiler is deprecated and we always use kineto, we will just make the tests use the new, kineto-only api